tests/test_query_model.py

```
@@ -228,3 +228,126 @@ async def test_unexpected_error_returns_model_error(self):
+            text="error",
+            request=httpx.Request("POST", "https://example.com"),
+        )
+        return httpx.HTTPStatusError("err", request=resp.request, response=resp)
+
+    @pytest.mark.asyncio
+    async def test_404_logs_at_warning_not_error(self, caplog):
+        """A 404 on a single model (e.g., deprecated endpoint) is gracefully handled."""
+        mock_post = AsyncMock(side_effect=self._make_status_error(404))
+
+        with patch.object(get_shared_client(), "post", mock_post):
+            with caplog.at_level("DEBUG", logger="backend.openrouter"):
```

Ruff (0.15.11)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)